File: dainf-lab-server/src/main/java/br/edu/utfpr/dainf/service/SubcategoryService.java
package br.edu.utfpr.dainf.service;

import br.edu.utfpr.dainf.model.Subcategory;
import br.edu.utfpr.dainf.repository.SubcategoryRepository;
import br.edu.utfpr.dainf.shared.CrudService;
import org.springframework.stereotype.Service;

@Service
public class SubcategoryService extends CrudService<Long, Subcategory, SubcategoryRepository> {
}

File: dainf-lab-client/src/app/pages/category/subcategory.service.ts
import { CrudService } from '@/shared/crud/crud.service';
import { Injectable } from '@angular/core';
import { Subcategory } from './subcategory';

@Injectable()
export class SubcategoryService extends CrudService<Subcategory> {
    constructor() {
        super('subcategories');
    }
}
